lib/src/util/cast_ui_factory.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:over_react/component_base.dart';
+
+/// Casts [value] to `UiFactory<T>`, which can be inferred from the surrounding context,
+/// allowing you to perform casts of dynamic values without having to repeat the type
+/// or worry about how formatting affects ignore comment placement.
+///
+/// Example:
+///
+/// ```dart
+/// // Explicit cast
+/// UiFactory<FooProps> Foo =
+///    // ignore: undefined_identifier
+///    _$Foo as UiFactory<FooProps>;
+///
+/// // This utility
+/// UiFactory<FooProps> Foo = castUiFactory(_$Foo); // ignore: undefined_identifier
```

Comment:
Would it be worth clarifying here that explicit casting is not the recommended syntax and that the utility should always be used to cast?

Reply:
Good catch, yep, I think so!